src/app/server/Server.cpp

```diff
@@ -314,15 +319,41 @@ static CHIP_ERROR OpenPairingWindowUsingVerifier(uint16_t discriminator, PASEVer
     params.SetPASEVerifier(verifier);
 #endif // CONFIG_NETWORK_LAYER_BLE
 
+    if (resetAdmins == ResetAdmins::kYes)
+    {
+        gNextAvailableAdminId = 0;
+        gAdminPairings.Reset();
+    }
+
+    AdminId admin                = gNextAvailableAdminId;
+    AdminPairingInfo * adminInfo = gAdminPairings.AssignAdminId(admin);
+    VerifyOrReturnError(adminInfo != nullptr, CHIP_ERROR_NO_MEMORY);
+    gNextAvailableAdminId++;
+
+    return gRendezvousServer.WaitForPairing(std::move(params), &gTransports, &gSessions, adminInfo);
+}
+
+static CHIP_ERROR OpenCertificatePairingWindow(ResetAdmins resetAdmins, OperationalCredentialSet * opCredSet)
+{
+    RendezvousParameters params;
+
+    params.SetOpCredSet(opCredSet);
+
+    if (resetAdmins == ResetAdmins::kYes)
+    {
+        gNextAvailableAdminId = 0;
```

Comment:
Why are we resetting the admins on CASE pairing? This will remove all previously paired admins on the device.